### ISSUE TITLE
'Cache dynamic channel queries' breaks pagination when using Structure

### ISSUE THREAD

**Issue by micahkohne** [ExpressionEngine/ExpressionEngine#2666]
**Description of the problem**
When using a listings channel on a template with pagination, the entries will not display correctly when navigating through the pages. This seems to be related to having 'Cache dynamic channel queries?' turned on.

**How To Reproduce**

1. Create template with a channel listing
2. Set channel to paginate
3. Turn on 'Cache dynamic channel queries?'
4. Cycling through the pages will not display the entries accurately

**Environment Details:**
 - Version: 7.2.5
 - PHP Version 7.4.21
 - MySQL Version 5.7.34
 - OS: macOS
 - Web Server: Apache

**Possible Solution**
Turning off 'Cache dynamic channel queries?' fixes the issue -- not sure if there is a fix that would allow this to work while remaining turned on.

**Comment by intoeetive** [comment]
@micahkohne it's been a while after you reported this, but are you still seeing the issue?

If yes, I wonder if you could share the template code that is problematic?

**Comment by micahkohne** [comment]
Yes @intoeetive -- I was working with eeharbor support and we were able to work out a fix. To get it to work, we had to do two things.

1. The page with the pagination has to be in a template group that matches the URL. For example, if I have a page at example.com/page-with-pagination, then I need to have a template group called 'page-with-pagination' and point the Structure entry to this template. Previously, I was pointing to a template that I had in a folder storing all of my Structure templates.
2. We had to shut off the extension 'core_template_route' by running UPDATE `exp_extensions` SET `enabled`='n' WHERE `class`='Structure_ext' AND `method`='core_template_route'
